ehrapy/api/io/read.py

```diff
@@ -264,33 +309,121 @@ def read_csv(
+            when multiple pdfs will be read, with multiple tables per file.
+
+            It's also important to note, that this has to be considered when passing "columns_obs_only":
+                .. code-block:: python
+                               import ehrapy.ap as ep
+                               # read pdf
+                               adata_dict = ep.io.read("my_tables.pdf", columns_obs_only={"0":["col1ofTable1", "col2OfTable1"],
+                               "1": ["colOfTable2"]})
+                               # this will put col1 and col2 of Table 0 of my_tables.pdf into obs only for this AnnData object
+                               # and col1 of Table 1 of my_tables.pdf into obs only for this respective AnnData object
+
+            Seems complicated at first glance, but this will allow the most flexibility for users.
+
+
+        Args:
+            filename: File path to the pdf.
+            index_column: Column name of the index column (obs)
+            columns_obs_only: Set of columns which only be stored in .obs, but not in X. Useful for free text annotations.
+            cache: Whether the data should be written to cache or not
+
+        Returns:
+            A dict of :class:`~anndata.AnnData` objects and the column obs only for each object
+        """
+        # possible extract modes are lattice (default) or stream; any of them may work better than the other one, depending on the data
+        pdf_extract_mode = kwargs.get("pdf_mode")
+        # camelot does not really parses headers in tables correctly; therefore, if there are any headers in the tables, set them as column names
+        header = kwargs.get("pdf_header")
+        # read a table list from the pdf
+        initial_df_list = camelot.read_pdf(
+            str(filename), flavor=pdf_extract_mode if pdf_extract_mode else "lattice", pages="all"
+        )
+        if not initial_df_list:
+            raise PdfParsingError(
+                f"Failed parsing file {filename}. Could not parse any data."
+                f"Consider converting your data files to .csv/.tsv before parsing or pass the "
+                f"guess parameter to the read function, which may improve parsed results!"
+            )
+
+        ann_data_objects = {}
+
+        # one pdf can contain multiple tables, so each of those tables will be one AnnData object
+        for idx, table in enumerate(initial_df_list):
+            df = table.df
+            is_set = isinstance(header, set)
+            # defaults to True, so if header has not been set, assume first row is column names row
+            # if header is a set and table number idx is in header, also assume first row is also column names row
+            if header is None or (is_set and idx in header):
+                # when the entry in top left corner is empty assume, table has header and index names stored in first row/first column
```

Comment:
I would add a yellow print as a warning when we do this.

Reply:
I'd say no. Imagine a user parses like 20 tables: Then he will get 20 print warnings since this should then print for all tables, since those could behave differently.